File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        int size = 0;
        ListNode temp = new ListNode();
        temp = head;
        while(temp != null){
            temp = temp.next;
            size++;
        }
        int n = (int)(Math.ceil(size / 2));
        ListNode curr= head;
        for(int i =0 ;i< n ;i++)
        {
            curr= curr.next;

        }
        return curr;
            
        }
        
    
};

File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    
    int getSize( ListNode *head)
    {
        int cnt = 0;
        ListNode * temp = head;
        while(temp != NULL)
        {
            temp = temp->next;
            cnt++;
        }
        return cnt;
        
    }
    
    ListNode* middleNode(ListNode* head) 
    {
        int size = getSize(head);
        cout<< size << endl;
        int mid = (size % 2 == 0) ? (size / 2 ) + 1: (size + 1) / 2;
        ListNode * curr = head;
        int cnt = 1;
        while(cnt != mid)
        {
            curr = curr->next;
            cnt++;
        }
        
        cout<< curr->val << endl;
        return curr;
        
        
    }
};